src/astra_driver.cpp

```diff
@@ -631,28 +955,43 @@ void AstraDriver::newIRFrameCallback(sensor_msgs::ImagePtr image)
 
 void AstraDriver::newColorFrameCallback(sensor_msgs::ImagePtr image)
 {
+  boost::mutex::scoped_lock lock(color_syncer_.mutex);
```

Comment:
looks better now. one last thing: can you move this lock in to the scope where you actually access color_mutex_.image? same for depthFrameCallback